### PR TITLE
<Tag/>, <MultiSelect/> - Added RTL support

### PR DESCRIPTION
**Pull request by rghorbani**
### What changed

Added RTL support to Tag and MultiSelect.

### Why it changed

Needed it ;)

---

- [x] Change is tested
- [x] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by hepiyellow** [comment]
Missing e2e tests 
- [ ] with tags (default - with remove button)
- [ ] with tags (with Thumb)

Optional improvements:
- I think we shouldn't have RTL examples in the story, it's creates clutter. I  think they should be in a Test story (under Tests category)
- We need to convert `<Tags/>` story to use AutoDocs, so we'll have a playground (in which you can play with RTL)

**Comment by hepiyellow** [comment]
@rghorbani Are you still into adding those e2e tests?

**Comment by rghorbani** [comment]
@hepiyellow Actually I had a little problem adding them. I'm not familiar with e2e test.

File: src/Tag/Tag.scss [hunk 1]
```diff
@@ -168,3 +168,88 @@
...
+
+    &.withRemoveButton {
+      padding-left: 3px;
+      padding-right: 9px;
+    }
+
+    .thumb {
+      margin-left: 6px;
+      margin-right: 0px;
+    }
```
[the remark this answers is not in the document]
Comment (rghorbani) [reply src/Tag/Tag.scss]: I responded according to the ltr code. In ltr .tinySize class does not have any implementation.

File: src/Tag/Tag.scss [hunk 2]
```diff
@@ -168,3 +168,88 @@
   white-space: nowrap;
   text-overflow: ellipsis;
 }
+
+@include withRTL() {
```
[the remark this answers is not in the document]
Comment (rghorbani) [reply src/Tag/Tag.scss]: As it is a great suggestion and I tested it it works really good but it has some inconsistencies with the ltr code like line 92 and 96.

File: src/Tag/Tag.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/Tag/Tag.scss]: you are right. because in tiny size, there is no margin between the text and the removeButton.

**Comment by hepiyellow** [comment]
Missing e2e snapshot tests. 
I’ll add them today.

File: stories/MultiSelect/index.story.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    dataHook: settings.dataHook,
+    value: '',
+    tags: [],
+    options,
+
+    valueParser,
+
+    onChange: e => setState({ value: e.target.value }),
+
+    onSelect: tags => {
```
Comment (hepiyellow) [inline stories/MultiSelect/index.story.js]: I don't understand this complex implementation of `onSelect.
The `ExampleStandard.js` implements : 
```js
handleOnSelect = tags =>
    Array.isArray(tags)
      ? this.setState({ tags: [...this.state.tags, ...tags] })
      : this.setState({ tags: [...this.state.tags, tags] });
```

Why is that implementation not good enough here?

**Comment by hepiyellow** [comment]
@rghorbani 
Since I will be adding the e2e tests, and I'd rather not push to this forked branch.
I suggest I will push the missing e2e tests in a separate PR, which is needed anyways in order to create a snapshot baseline for missing tests.

Then you can merge (update) your branch master, and we can merge it.

File: stories/MultiSelect/index.story.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      setState({
+        tags: getState().tags.filter(currTag => currTag.id !== tagId),
+      }),
+  }),
+
+  exampleProps: {
+    // onSelect: option => option.value,
+    // onChange: e => e.target.value,
+    options: [
+      { label: 'One option', value: [{ id: 'WSR', value: 'WSR' }] },
```
Comment (hepiyellow) [inline stories/MultiSelect/index.story.js]: Why do we need this `One option` example?

File: stories/MultiSelect/index.story.js (hunk 3 above)
Comment (rghorbani) [reply stories/MultiSelect/index.story.js]: I clarified with comments.
Also checking if option exist before adding for manual entry (avoiding conflicts).

File: stories/MultiSelect/index.story.js (hunk 4 above)
Comment (rghorbani) [reply stories/MultiSelect/index.story.js]: Removed.

**Comment by hepiyellow** [comment]
@rghorbani 
I merged this PR #2505
which adds the relevant e2e tests for MultiSelect and Tag, including RTL.

Could you rebase your branch, and we'll see how the snapshots comparison looks.

**Comment by rghorbani** [comment]
Thank you @hepiyellow. I merged the code and everything seems to be fine.

**Comment by hepiyellow** [comment]
@rghorbani Our e2e snapshot testing are disabled temporarily (some timeout issues),
So I'll wait until tomorrow, to see if the issues are fixed.
If not, I'll merge it, and we'll confirm the snapshot baseline later when e2e tests are enabled.

**Comment by hepiyellow** [comment]
@milkyfruit @LittleTal 
Can you confirm the new RTL fixes for `<Tags/>` (`<MultiSelect/>`) and `<Tag/>`?

This is the [Applitools PR branch](https://eyes.applitools.com/app/test-results/00000251858648576333/00000251858647533088/steps/1?accountId=mE99eP8CUn9UTSXSv98L3OSaJ0NZmAcY6Wndx6WSHODdY110) where you can Confirm new snapshot baselines.

This is the storybook for [`<Tags/>`](https://wix-wix-style-react-pr-2356.surge.sh/?selectedKind=3.%20Inputs&selectedStory=3.8%20Tags&full=0&addons=0&stories=1&panelRight=0) and [`<Tag/>`](https://wix-wix-style-react-pr-2356.surge.sh/?selectedKind=12.%20Other&selectedStory=12.5%20Tag&full=0&addons=0&stories=1&panelRight=0)

You can use the Playground's "imitateRTL" to verify it.

- For `<Tags/>` there is a new RTL margin for a tag in the input box.
- For `<Tag/>` there are new padding/margins for `removeButton` and `thumb` in all sizes.

**Comment by hepiyellow** [comment]
@LittleTal 
I think you meant to attach this image:
screen shot 2018-12-03 at 14 06 13
 
Which corresponds to sizes: tiny, small, medium , large.

**Comment by rghorbani** [comment]
@hepiyellow Is there any work need to be done?

**Comment by rghorbani** [comment]
@hepiyellow fixed it and checked it visually.

**Comment by mykas** [comment]
Guys, please revert this because my PR that updates CloseButton https://github.com/wix/wix-style-react/pull/2480 was released 5 minutes before your PR. There might be some changes you need to do.

**Comment by hepiyellow** [comment]
@mykas
Did the CloseButton change any right/left margin/padding?

I see that the change you made to Tag is in master, and it looks fine to me.

**Comment by mykas** [comment]
Actually no, it did not do anything to those. I just remapped props.

**Pull request merged**